Homework_11_Java_Circus(Exceptions_Generics_Annotations)/Homework_11_Java_Circus/src/CircusPerformer.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Represents a circus performer.
+ */
+public class CircusPerformer {
+	private String name;
+	private String act;
+	private int experience;
+	private String experienceMessage;
+
+	public String getName() {
+		return name;
+	}
+
+	public String getAct() {
+		return act;
+	}
+
+	public int getExperience() {
+		return experience;
+	}
+
+	public String getExperienceMessage() {
+		return experienceMessage;
+	}
+
+	/**
+	 * Exception class for invalid experience values.
+	 */
+	public static class InvalidExperienceException extends Exception {
+		public InvalidExperienceException(String message) {
+			super(message);
+		}
+	}
+
+	public CircusPerformer(String name, String act, int experience) {
+		this.name = name;
+		this.act = act;
+		setExperience(experience);
+	}
+
+	/**
+	 * Sets the experience of the performer, updating the experience message.
+	 */
+	private void setExperience(int experience) {
+		try {
+			if (experience <= 0) {
+				throw new InvalidExperienceException("First experience - then circus!");
+			} else if (experience > 50) {
+				throw new InvalidExperienceException("Time to retire!");
+			} else {
+				this.experience = experience;
+				this.experienceMessage = experience + " years of experience.";
```

Comment:
Maybe I didn't get it, but do we really need to have a separate field for this and construct an `experienceMessage` here instead of just having a method which concatenates the value of stringified `experience` field and `" years of experience"?

I mean an `experience` is a concrete attribute of a performer, though `a message that says smth about experience` can be generated on-the-fly, right?